tests/tck-build-logic/src/main/java/org/graalvm/internal/tck/ConfigFilesChecker.java

```diff
@@ -0,0 +1,399 @@
+package org.graalvm.internal.tck;
+
+import groovy.json.JsonSlurper;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.options.Option;
+
+import java.util.Scanner;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.stream.Collectors;
+
+/**
+ * Checks content of config files for a new library.
+ * <p>
+ * Run with {@code gradle checkConfigFiles --coordinates com.example:library:1.0.0}.
+ */
+public abstract class ConfigFilesChecker extends DefaultTask {
+
+    @InputFiles
+    protected abstract RegularFileProperty getMetadataRoot();
+
+    @InputFiles
+    protected abstract RegularFileProperty getIndexFile();
+
+    private final Set<String> EXPECTED_FILES = new HashSet<>(Arrays.asList("index.json", "reflect-config.json", "resource-config.json", "serialization-config.json",
+            "jni-config.json", "proxy-config.json", "predefined-classes-config.json"));
+
+    private final Set<String> ILLEGAL_TYPE_VALUES = new HashSet<>(List.of("java.lang"));
+
+    private final Set<String> PREDEFINED_ALLOWED_PACKAGES = new HashSet<>(Arrays.asList("java.lang", "java.util"));
+
+    Coordinates coordinates;
+    List<String> allowedPackages;
+
+    @Option(option = "coordinates", description = "Coordinates in the form of group:artifact:version")
+    void setCoordinates(String coords) {
+        this.coordinates = Coordinates.parse(coords);
+
+        File coordinatesMetadataRoot = getProject().file(CoordinateUtils.replace("metadata/$group$/$artifact$/$version$", coordinates));
+        getMetadataRoot().set(coordinatesMetadataRoot);
+
+        File index = getProject().file("metadata/index.json");
+        getIndexFile().set(index);
+
+        this.allowedPackages = getAllowedPackages();
+    }
+
+
+    @TaskAction
+    void run() throws IllegalArgumentException, FileNotFoundException {
+        if (coordinates.group().equalsIgnoreCase("org.example") || coordinates.group().equalsIgnoreCase("samples")) {
```

Comment:
Perhaps we could also move these two libraries into a "library names that we don't check" list